File: sdkwork-backend-sdk-flutter/lib/src/api/ai_image_generation.dart
import '../http/client.dart';
import '../models.dart';

class AiImageGenerationApi {
  final HttpClient _client;
  
  AiImageGenerationApi(this._client);

  /// Create image generation task
  Future<PlusApiResultGenerateImageVO?> create(GenerateImageForm body) async {
    final response = await _client.post(ApiPaths.backendPath('/generation/image/create'), body: body, contentType: 'application/json');
    return response is PlusApiResultGenerateImageVO ? response : null;
  }

  /// Get image generation result
  Future<PlusApiResultGenerateImageVO?> getResult(String taskId) async {
    final response = await _client.get(ApiPaths.backendPath('/generation/image/result/${taskId}'));
    return response is PlusApiResultGenerateImageVO ? response : null;
  }
}

File: sdkwork-backend-sdk-go/api/ai_image_generation.go
package api

import (
    "fmt"
    sdktypes "github.com/sdkwork/backend-sdk/types"
    sdkhttp "github.com/sdkwork/backend-sdk/http"
)

type AiImageGenerationApi struct {
    client *sdkhttp.Client
}

func NewAiImageGenerationApi(client *sdkhttp.Client) *AiImageGenerationApi {
    return &AiImageGenerationApi{client: client}
}

// Create image generation task
func (a *AiImageGenerationApi) Create(body sdktypes.GenerateImageForm) (sdktypes.PlusApiResultGenerateImageVO, error) {
    raw, err := a.client.Post(BackendApiPath("/generation/image/create"), body, nil, nil, "")
    if err != nil {
        var zero sdktypes.PlusApiResultGenerateImageVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultGenerateImageVO](raw)
}

// Get image generation result
func (a *AiImageGenerationApi) GetResult(taskId string) (sdktypes.PlusApiResultGenerateImageVO, error) {
    raw, err := a.client.Get(BackendApiPath(fmt.Sprintf("/generation/image/result/%s", taskId)), nil, nil)
    if err != nil {
        var zero sdktypes.PlusApiResultGenerateImageVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultGenerateImageVO](raw)
}
